hooks/usePayment.tsx

```diff
@@ -12,24 +10,22 @@ const usePayment = () => {
     isLoading: isLoadingCredits,
     refetch: refetchCredits,
   } = useCredits();
-  const { data: subscriptionData, isLoading: isLoadingSubscription } =
-    useSubscription();
+  const { data: proStatusData, isLoading: isLoadingProStatus } = useProStatus();
 
   const isLoadingUser = email === undefined || (!!email && !userData);
   const credits = creditsData?.remaining_credits || 0;
-  const subscription = subscriptionData?.subscription;
 
-  const subscriptionActive =
-    isEnterprise(email || "") || isActiveSubscription(subscription);
+  // Check pro status (account subscription or org subscription)
+  const subscriptionActive = proStatusData?.isPro || false;
   const totalCredits = subscriptionActive ? PRO_CREDITS : DEFAULT_CREDITS;
 
   return {
-    isLoading: isLoadingCredits || isLoadingSubscription || isLoadingUser,
+    isLoading: isLoadingCredits || isLoadingProStatus || isLoadingUser,
     credits,
     totalCredits,
     subscriptionActive,
     refetchCredits,
-    subscription,
+    proSource: proStatusData?.proSource,
```

Comment:
YAGNI principle
- actual: proSource is not used anywhere from the hook.
- required: delete unused variables.